### ISSUE TITLE
SD read but RomWBW not detecting IDE0

### ISSUE THREAD

**Issue by zomgugoff** [ExtremeElectronics/Pico-SD--IDE-interface--for-Rc2014#2]
This is a follow-up on the post I made in the RomWBW repo. The IDE device is detected, but IDE0 and IDE1 are not. I read out the serial data from the Pico while it booted and it didn't report any errors. Below is the output. It specifies that IDE0 is open, but RomWBW says "NO MEDIA" for IDE0. When booting the RC2014, IO is logged also.

sd_spi_go_low_frequency: Actual frequency: 400000
V2-Version Card
R3/R7: 0x1aa
R3/R7: 0xff8000
R3/R7: 0xc0ff8000
Card Initialized: High Capacity Card
SD card initialized
SDHC/SDXC Card: hc_c_size: 29567
Sectors: 30277632
Capacity:    14784 MB
sd_spi_go_high_frequency: Actual frequency: 12000000

SD INIT OK 

Ini file rc2014.ini Exists Loading ... 

Loaded INI

IDE0 Open OK

Trace:1

 OK lets start the z80 stuff

 Listning for IDE activity... 

Start?:

OK

io:1C00 Wr reg[07]->40 064

io:1C00 Wr reg[07]->40 064

io:0C00 Wr reg[03]->01 001

io:1C00 Wr reg[07]->40 064

io:0C00 Wr reg[03]->01 001

**Comment by ExtremeElectronics** [comment]
Sorry, but as I mentioned on the WBW issues, I have no way of testing this I don't have a machine capable of running WBW with the interface.

**Comment by ExtremeElectronics** [comment]
Does the SD card interface work with a stock RC2014 build ?

**Comment by wwarthen** [comment]
Hi @zomgugoff,

Could you please post your rc2014.ini file?  I will try to replicate your issue.

Thanks, Wayne

**Comment by zomgugoff** [comment]
I have not tried a stock build. I can try it. I will post my INI when I get home. I can say it's unchanged from the WBW INI except for fixing the image filename (because the archived filename is slightly different)

**Comment by zomgugoff** [comment]
@wwarthen Here is my INI

```
#
# RC2014 ROM WBW configuration #
#

[IDE]
#file to use for IDE drives;
idefilei = "RomWBWIDE.id";
iscf = 0;
idefile = "hd_combo-RC2014 Zed.img";

[DEBUG]
trace=1;


#for other settings see README

```

**Comment by wwarthen** [comment]
@zomgugoff, your rc2014.ini matches mine (except I don't have trace enabled).  This configuration is working for me.  I'm not sure what to suggest.

-Wayne

**Comment by zomgugoff** [comment]
Do you have any output with trace enabled I can look at? I would be curious to see what I should be seeing. Also, if I were to make my image file available, would that be worth testing?

**Comment by zomgugoff** [comment]
@ExtremeElectronics I only have the 512K RAM/ROM board. I don't think I can run stock ROMs. But in the most minimal config, It's still reporting no media.

**Comment by wwarthen** [comment]
@zomgugoff, I captured this log file showing the initial power-up of the board followed by RomWBW booting.

[boot.log](https://github.com/user-attachments/files/16257912/boot.log)

Thanks, Wayne

**Comment by zomgugoff** [comment]
Aha, I see only writes in my output whereas you have reads and writes. I'm afraid I don't know enough about the Pico nor the RC2014 to discern what is to/from on those lines, but obviously, I've only got 1-way communication. Thank you for doing that.

**Comment by zomgugoff** [comment]
I just noticed I've started getting a 
```ide: cf: 0: bad data read```
Error in my output. Not sure if that indicates the read from the RC2014 was bad or the read from the SD was bad.

**Comment by zomgugoff** [comment]
And now, I will reveal a simple case of user error:

I started re-poking the solder joints and found a short between GPIO14 and GPIO15. In the process of tracking this down, I attempted to remove RN2 (where I initially found the short) and broke pins 2-6 off. They were still in the board, so I reattached the remaining legs and at the very least, IDE0 is now detected.

I apologize for the run-around in tracking this issue down. I now have to make an 'order of shame' for a new resistor network.

Thanks again for putting up with my self-inflicted issue.
